src/main/CMakeLists.txt

```diff
@@ -36,16 +36,18 @@ set_source_files_properties(${libmain_gen_src_files} PROPERTIES GENERATED TRUE)
 
 add_library(main OBJECT ${libmain_src_files} ${libmain_gen_src_files})
 target_compile_definitions(main PRIVATE -D__BUILDING_CVC5DRIVER)
-if(ENABLE_SHARED)
-  set_target_properties(main PROPERTIES POSITION_INDEPENDENT_CODE ON)
+set_target_properties(main PROPERTIES POSITION_INDEPENDENT_CODE ON)
+
+if(ENABLE_STATIC_BINARY)
+  target_link_libraries(main PUBLIC cvc5-static cvc5parser-static)
+else()
+  target_link_libraries(main PUBLIC cvc5-shared cvc5parser-shared)
+endif()
+if(USE_CLN)
+  target_link_libraries(main PUBLIC CLN)
 endif()
```

Comment:
Do we still have to link against CLN separately here? If so, why?

Reply:
No, and we don't need GMP either.